File: src/53-maximum-subarray/solution.rb
# @param {Integer[]} nums
# @return {Integer}
def max_sub_array(nums)
    res = nums[0]
    sum = 0
    nums.each do
        sum = [0, sum].max
        sum += _1
        res = [res, sum].max
    end
    res
end

File: src/53-maximum-subarray/solution.ts
function maxSubArray(nums: number[]): number {
  let res: number = nums[0], sum: number = 0;
  for (const num of nums) {
    sum = Math.max(sum, 0);
    sum += num;
    res = Math.max(res, sum);
  }
  
  return res;
};
